ArmRobot/Assets/Scripts/RandomizerPositionObject.cs

```diff
@@ -0,0 +1,165 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class RandomizerPositionObject : MonoBehaviour
+{
+    private GameObject table;
+
+    private GameObject gameobjectSeen;
+    float robotMinReach = 0.2f;
+    float robotMaxReach = 0.5f;
+
+    Bounds tableBounds;
+    float targetY;
+
+
+    void Start()
+    {
+        //GameObject[] arrayofspheres = GameObject.FindGameObjectsWithTag("Sphere");
+        table = GameObject.Find("Table");
+        tableBounds = table.GetComponent<Collider>().bounds;
+        targetY = transform.position.y;
+    }
+
+
+    // CONTROL
+
+    public void Move(List<GameObject> listOfAlreadyMovedObjects)
+    {
+        // random position (on table, within reach)     
+        Vector2 tableTopPoint = RandomReachablePointOnTable(listOfAlreadyMovedObjects);
+        Vector3 tableCenter = tableBounds.center;
+        float x = tableCenter.x + tableTopPoint.x;
+        float z = tableCenter.z + tableTopPoint.y;
+        transform.position = new Vector3(x, targetY, z);
+
+        // random rotation
+        Vector3 randomRotation = new Vector3(
+            transform.rotation.eulerAngles.x,
+            Random.value * 360.0f,
+            transform.rotation.eulerAngles.z);
+        transform.rotation = Quaternion.Euler(randomRotation);
+    }
+
+
+    // HELPERS
+
+    Vector2 RandomReachablePointOnTable(List<GameObject> listOfAlreadyMovedObjects)
+    {
+        while (true)
+        {
+            Vector2 randomOffset = RandomPoint(robotMinReach, robotMaxReach, listOfAlreadyMovedObjects);
+            bool onTable = PointOnTable(randomOffset);
+            if (onTable)
+            {
+                return randomOffset;
+            }
+        }
+    }
+
+    bool PointOnTable(Vector2 point)
+    {
+        /*  point: The 2D point on table top, relative to center of table top.
+         *  Determines if this point would be on the table or not.      
+         */
+        Vector3 tableExtents = tableBounds.extents;
+        float targetRadius = 0f;
+        if (tag == "Sphere"){
+                targetRadius = GetComponent<SphereCollider>().radius * GetComponent<Transform>().localScale[0]; // for the sphere 
+            }
+            // otherwise it is a cylinder 
+            else{
+                targetRadius = GetComponent<CapsuleCollider>().radius * GetComponent<Transform>().localScale[0]; 
+            }
+
+        float safeXDistance = tableExtents.x - targetRadius;
+        float safeZDistance = tableExtents.z - targetRadius;
+        float xDistance = Mathf.Abs(point.x);
+        float yDistance = Mathf.Abs(point.y);
+        bool onTable = (xDistance < safeXDistance) && (yDistance < safeZDistance);
+        return onTable;
+    }
+
+    public Vector2 RandomPoint(float minRadius, float maxRadius, List<GameObject> listOfAlreadyMovedObjects)
```

Comment:
Is this method called from anywhere outside this file? If not, it does not need to be `public`.

Reply:
Also, this method is rather long and hard to read at a glance. I think if you moved some of this out into a separate method like `bool PositionIsValid(Vector2 randomPoint, List<GameObject> listOfAlreadyMovedObjects)` you could greatly improve the readability.